pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -574,35 +576,96 @@ func (ssc *defaultStatefulSetControl) updateStatefulSet(
 		return &status, nil
 	}
 
+	return updateStatefulSetAfterInvariantEstablished(ctx,
+		ssc,
+		set,
+		replicas,
+		updateRevision,
+		status,
+	)
+
+}
+
+func updateStatefulSetAfterInvariantEstablished(
+	ctx context.Context,
+	ssc *defaultStatefulSetControl,
+	set *apps.StatefulSet,
+	replicas []*v1.Pod,
+	updateRevision *apps.ControllerRevision,
+	status apps.StatefulSetStatus,
+) (*apps.StatefulSetStatus, error) {
+
+	replicaCount := int(*set.Spec.Replicas)
+
 	// we compute the minimum ordinal of the target sequence for a destructive update based on the strategy.
 	updateMin := 0
+	maxUnavailable := 1
+	var err error
```

Comment:
Declare `err` right before it's needed, too easy to accidentally reference it if it is declared here.

Reply:
yes its used right below, do you instead prefer := syntax to allocate it everytime ?

Reply:
> too easy to accidentally reference it if it is declared here

Please do not make it easy to accidentally reference a variable that was set within a different scope for an unrelated purpose.